### ISSUE TITLE
Enhancement: remove inside-viewport screenshot limitation.

### ISSUE THREAD

**Issue by garris** [OnetapInc/chromy#15]
I have a major use-case where I will need to capture selectors which are longer or wider than the viewport.   Do you have any suggestions on how this could be implemented?

**Comment by dotneet** [comment]
I have some ideas that makes it possible.
I change the implementation of screenshotSelector() to eliminate the limitation.

**Comment by garris** [comment]
Much appreciated!

**Comment by garris** [comment]
Just a thought.  Today I integrated chromy.screenshotDocument() -- this feature works very well!   A good side-effect is that the scroll bar disappears.  For screenshotSelctor() you could probably just run screenshotDocument() and then crop the resulting picture with values from getBoundingClientRect().  

Imagine however the performance benefit of this approach, screenshotSelctor(["#myBanner",  "#myForm", "#myNewsItem", "document"]) -- this operation, which used to require multiple renderings would only require one screenshot.  Many backstopjs users will run hundreds of tests at a time -- this can take a very long time to complete -- so anywhere speed can be optimized is critical.

Also, currently, backstop enables selctorExpansion e.g...
{
  selectors: ['img'],
  selctorExpansion: true
}
The above will first `querySelectorAll(selector)` and then take screenshots for all `<img>` on the page.  note: `selctorExpansion = false` will do `querySelector(selector)` -- IOW only first occurrence.

You may want to consider creating a methods like `screenshotSelctors(selectorsArray, doExpansion)`   the speed from this would be a big benefit for users.  

These are just suggestions.  Not requests.  I know you will be busy with a lot of requests as more users contact you with questions.

**Comment by garris** [comment]
I just read this article -- seems the author seems to address the "known issue -- strange white area" issue you had noted?

https://medium.com/@dschnr/using-headless-chrome-as-an-automated-screenshot-tool-4b07dffba79a

**Comment by dotneet** [comment]
Thanks great suggestion!
I consider to implement something similar to that.

I have already read that article. The known issue is different problem.
The problem is reproduced in his repository. (https://github.com/schnerd/chrome-headless-screenshots)
I'm aware of that what causes white area problem.
It's a Chrome's issue.
headless chrome59 needs --disable-gpu option to start up because of bug.
disable-gpu option is a reason for known issue and no need in v.60 or later.
thus, after chrome60 is released officially this problem will be resolved.

**Comment by garris** [comment]
Ahh, I see -- Makes sense -- thanks for the explanation!

**Comment by dotneet** [comment]
By the way, I'm busy in this week.
Probably I can't implement additional functions until this weekend or next week.

**Comment by garris** [comment]
No worries!  We are on holiday starting this saturday for a week.   Enjoy your time off!

**Comment by dotneet** [comment]
I added screenshotMultipleSelectors() 2 days ago.
https://github.com/OnetapInc/chromy#screenshotmultipleselectorsselectors-callback-options--

The command has useQuerySelectorAll option.
It correspond to selectorExpansion.

It also resolves #14.

**Comment by garris** [comment]
Wow!  I am very excited to try this out!   Thank you for adding this feature!

**Comment by garris** [comment]
screenshotMultipleSelectors() is a good idea -- but I am having some trouble working with it.  

First, I would expect that screenshotMultipleSelectors() would have similar usage pattern to the other screenshot() commands.  

other command patterns...
```
.screenshotSelector('.selector')
.result(singleDataProperty => {
   something synchronous here
})
```

I would expect this pattern...
```
.screenshotMultipleSelectors(['.selector1','.selector2'])
.result([arrayOfMultipleDataProperties] => {
   synchronous sequential process data here
})
```

Also, since I need to maintain compatibility with currently supported node I don't want to add `async` commands to my code.   I would prefer if I could return a promise inside of .result() such that chromy would wait to resolve to the next chain until my promise has resolved.  (this would allow me to do multiple saves asynchronously)

Mapping the `dataArrayOfObjects` to a `Promise.all(array of async api methods)` seems more intuitive here with respect to backward compatibility.

I don't know if this is possible or something that you would agree with.  Would like to know your thoughts.  Thanks!

**Comment by dotneet** [comment]
What if a promise returned by callback of screenshotMultipleSelectors() is resolved? Would your problem be resolved?

```js
.screenshotMultipleSelectors(['.sel1', '.sel2'], function(err, image) {
  return new Promise( function(resolve, reject) {
    // process an image
  })
})
.someCommand() // this command will be run after all images has processed.
```

In any case a promise returned by callback should be resolved by screenshotMultipleSelectors().
So I've updated chromy.

I designed screenshotMultipleSelectors()  for reducing memory usage.
If a lot of selectors passed in the command, this design is required.
